src/metatrain/utils/data/dataset.py

```diff
@@ -230,6 +196,133 @@ def union(self, other: "DatasetInfo") -> "DatasetInfo":
         return new
 
 
+class Dataset:
+    """A version of the `metatensor.learn.Dataset` class that allows for
+    the use of `mtm::` prefixes in the keys of the dictionary. See
+    https://github.com/lab-cosmo/metatensor/issues/621.
```

Comment:
Perhaps not for this PR, but I think this is merged now so should allow for these prefixes

Reply:
This is not yet released, but we should do a patch release with this